modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgnitePlanner.java

```diff
@@ -560,6 +564,11 @@ public void checkCancel() {
                 long startTs = ctx.startTs();
 
                 if (FastTimestamps.coarseCurrentTimeMillis() - startTs > timeout) {
+                    LOG.info("Planning of a query aborted due to planner timeout threshold is reached [timeout={}, query={}]",
```

Comment:
I'm not sure we should log it at INFO level

Reply:
Maybe it is better to move logging to Context, because it contains all the required information for logging, and the planner will not bother about what should be logged.

Reply:
the same: conversation is resolved, but nothing has changed

Reply:
> But I don't see right now proper way to inform user about the situation

the user will be informed by getting an exception

Reply:
It's not true. We can have two cases:
1) was have cancel planning by timeout, however already there is correct plane to execute.
2) Timeout reached, but we don't have a plan at all.

user get exception just in the second case. I think user should have ability be aware about the first case too.

Reply:
> I think user should have ability be aware about the first case too

sounds like we need to expose metric for this